apps/coscrad-frontend/src/components/tree-of-knowledge/category-tree.presenter.tsx

```diff
@@ -1,50 +1,36 @@
-import { AggregateType, CategorizableType, ICategoryTreeViewModel } from '@coscrad/api-interfaces';
-import { Accordion, AccordionDetails, AccordionSummary, Typography } from '@mui/material';
-import { buildDataAttributeForAggregateDetailComponent } from '../../utils/generic-components/presenters/detail-views/build-data-attribute-for-aggregate-detail-component';
+import { CategorizableType, ICategoryTreeViewModel } from '@coscrad/api-interfaces';
+import { List, Typography } from '@mui/material';
 import { FunctionalComponent } from '../../utils/types/functional-component';
-import { CategorizablesOfMultipleTypeContainer } from '../higher-order-components';
-import { thumbnailCategorizableDetailPresenterFactory } from '../resources/factories/thumbnail-categorizable-detail-presenter-factory';
+import { CategoryTreeNode } from './category-tree-node';
+import { CategoryTreeUXProvider } from './category-tree-ui-context';
 
 const wrapTree = ({
     id,
     label,
     children,
     members,
-}: ICategoryTreeViewModel<CategorizableType>): JSX.Element => (
-    <div
-        data-testid={buildDataAttributeForAggregateDetailComponent(AggregateType.category, id)}
-        key={buildDataAttributeForAggregateDetailComponent(AggregateType.category, id)}
-    >
-        <Accordion>
-            <AccordionSummary>{label}</AccordionSummary>
-            <AccordionDetails>
-                {/* TODO Pass the state in as props */}
-                {/* TODO Use `SelectedCategorizablesOfMultipleTypesPresenter` as in `Notes` and `Tags` flow */}
-                <CategorizablesOfMultipleTypeContainer
-                    members={members}
-                    detailPresenterFactory={thumbnailCategorizableDetailPresenterFactory}
-                />
-                {
-                    // recurse on the children
-                    children.map(wrapTree)
-                }
-            </AccordionDetails>
-        </Accordion>
-    </div>
-);
+}: ICategoryTreeViewModel<CategorizableType>): JSX.Element => {
+    return (
+        <CategoryTreeNode id={id} label={label} members={members} wrapTree={wrapTree}>
```

Comment:
I find passing wrapTree as a prop a bit hard to reason about. Is there a better pattern we can use?